tests/vmlifecycle_test.go

```diff
@@ -159,6 +159,32 @@ var _ = Describe("Vmlifecycle", func() {
 			})
 		})
 
+		Context("with boot order", func() {
+			It("should be able to boot from 'middle' disk", func() {
+				By("defining a VM with an Alpine disk")
+				vm = tests.NewRandomVMWithEphemeralDisk(tests.RegistryDiskFor(tests.RegistryDiskAlpine))
+				By("adding a Cirros Disk with boot order 1")
+				tests.AddEphemeralDiskWithBootOrder(vm, "disk1", "virtio", tests.RegistryDiskFor(tests.RegistryDiskCirros), 1)
+				By("adding a Fedora Disk")
+				tests.AddEphemeralDisk(vm, "disk2", "virtio", tests.RegistryDiskFor(tests.RegistryDiskFedora))
+
+				By("starting VM")
+				vm.Spec.Hostname = "cirros"
```

Comment:
Is it necessary to set the hostname?

Reply:
It looks to be required to me. The LoggedInCirrosExpecter looks for 'vmName + " login:"', which seems to be 'cirros login:', and vmName is either from vm.Name (which is dynamically generated in these tests) or vm.Spec.Hostname.  I'll admit that I looked at other uses of this and it wasn't immediately clear to me how those were getting past this.

Reply:
I ditched this anyway and wrote an expecter function that can be used for the multiple OS tests.